src/ray/gcs/gcs_server/gcs_table_storage.cc

```diff
@@ -139,17 +153,15 @@ Status GcsTableWithJobId<Key, Data>::BatchDelete(const std::vector<Key> &keys,
     keys_to_delete.push_back(key.Binary());
   }
   return this->store_client_->AsyncBatchDelete(
-      this->table_name_, keys_to_delete, [this, callback, keys](auto status) {
-        if (status.ok()) {
-          {
-            absl::MutexLock lock(&mutex_);
-            for (auto &key : keys) {
-              index_[GetJobIdFromKey(key)].erase(key);
-            }
+      this->table_name_, keys_to_delete, [this, callback, keys](auto) {
```

Comment:
```suggestion
      this->table_name_, keys_to_delete, [this, callback=std::move(callback), keys=std::move(keys)](auto) {
```
Nit!

Reply:
nit: just move won't work here since callback and keys are constant. A rule to follow is to pass by value if we'd like to move the data around. (https://stackoverflow.com/questions/51705967/advantages-of-pass-by-value-and-stdmove-over-pass-by-reference)

Because this gives the caller the ability to avoid copying.